pup-0002.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Docstring Style Conventions
+author: Austin Macdonald
+created: 28-Mar-2017
+status: proposed
+---
+
+## Summary
+
+This PUP documents docstring style conventions for Pulp 3.0.
+
+## Motivation
+
+To improve docstring readability and increase information density.
+
+## Details
+
+Upon acceptance, these changes will be documented in the Pulp style guide.
+
+### Style Choices
+
+Docstrings should generally follow the conventions provided in the comments section of the
+[Google style guide.](http://google.github.io/styleguide/pyguide.html?showone=Comments#Comments)
+
+Additionally, docstrings must be parsable by the
+[Napoleon plugin for Sphinx.](http://www.sphinx-doc.org/en/stable/ext/napoleon.html)
+
+### Additions/Exceptions
+
+Pulp conventions that differ from the `Comments` section of the Google style guide will be
+explicitly included in the Pulp style guide.
+
+1. Modules will not contain license information. Our license information is presented correctly
+within the project; it is unnecessary to include it in each file.
+
+2. The type of each `Arg` value should be included after the variable name in parentheses. The type of each `Returns` value should be the first item on the line. 
+
+3. Following the type of `Arg` and `Return` values, there will be a colon and a single space
+followed by the description.
+
```

Comment:
Additionally, we also have added some django-related sections to our docs handler in 3.0, "Fields" and "Relations", used to describe fields and relations defined on models. This was determined to be needed because of the difference between fields on model classes and model instances. On model classes, the value of the field is the field class itself. On model instances, the value of the field is an instance of the field that contains its DB value, or a manager that allows querying across the relation that the field represents. As a result, it's not entirely appropriate to refer to model fields/relations as attributes (either class atrributes or instance attributes).

I propose something like:
```
4. `Fields` and `Relations` sections will be used when documenting fields on Django models. The `Fields` section will be used for non-related fields on Model classes, the `Relations` section will be used for related fields on Model classes.
```

If needed, we can reference the Django docs to help determine what fields are related:
https://docs.djangoproject.com/en/1.8/ref/models/fields/#module-django.db.models.fields.related